core/src/test/java/org/apache/iceberg/TestEntriesMetadataTable.java

```diff
@@ -68,20 +63,19 @@ public void testEntriesTableScan() {
     Schema expectedSchema =
         TypeUtil.join(readSchema, MetricsUtil.readableMetricsSchema(table.schema(), readSchema));
 
-    assertEquals(
-        "A tableScan.select() should prune the schema",
-        expectedSchema.asStruct(),
-        scan.schema().asStruct());
+    assertThat(scan.schema().asStruct())
+        .as("A tableScan.select() should prune the schema")
+        .isEqualTo(expectedSchema.asStruct());
 
     FileScanTask file = Iterables.getOnlyElement(scan.planFiles());
-    Assert.assertEquals(
-        "Data file should be the table's manifest",
-        Iterables.getOnlyElement(table.currentSnapshot().allManifests(table.io())).path(),
-        file.file().path());
-    Assert.assertEquals("Should contain 2 data file records", 2, file.file().recordCount());
+    assertThat(file.file().path())
+        .as("Data file should be the table's manifest")
+        .isEqualTo(
+            Iterables.getOnlyElement(table.currentSnapshot().allManifests(table.io())).path());
```

Comment:
can we express this check without using `Iterables`?

Reply:
Sure, replace it with `get(0)`.